src/components/radio-button-group/RadioButtonGroup.tsx

```diff
@@ -0,0 +1,154 @@
+import styled from '@emotion/styled';
+import { ReactNode } from 'react';
+
+interface Props {
+  onChange?: (value: string) => void;
+  options: RadioButtonProps[];
+
+  // Group Props
+  defaultValue?: string;
+  disabled?: boolean;
+  id?: string;
+  required?: boolean;
+  value?: string;
+}
+
+export function RadioButtonGroup(props: Readonly<Props>) {
+  const { onChange, options, ...groupProps } = props;
+
+  return (
+    <RadioGroupWrapper onValueChange={onChange} {...groupProps}>
+      {options.map((o) => (
+        <RadioButton groupId={groupProps.id} {...o} />
+      ))}
+    </RadioGroupWrapper>
+  );
+}
+
+/*
+ * Related to the explanation below, we require an aria-label if the label is a ReactNode.
+ * If the label is a string, we use it directly as a fallback if no aria-label is provided
+ */
+type RadioButtonProps = {
+  disabled?: boolean;
+  groupId?: string;
+  value: string;
+} & ({ ariaLabel: string; label: ReactNode } | { ariaLabel?: string; label: string });
+
+function RadioButton(props: Readonly<RadioButtonProps>) {
+  const { ariaLabel, disabled, groupId = 'radio', label, value } = props;
+
+  const id = `${groupId}-${value}`;
+
+  /*
+   * Although the HTML spec defines buttons as valid targets for labels,
+   * some Screen Readers have issues with those:
+   * https://github.com/dequelabs/axe-core/issues/3696#issuecomment-1665575742
+   * So here we force an aria-label:
+   */
+  const inputLabel = ariaLabel ?? (typeof label === 'string' ? label : undefined);
+
+  return (
+    <OptionWrapper>
+      <Input aria-label={inputLabel} disabled={disabled} value={value} id={id}>
+        <SelectionIndicator />
+      </Input>
+      <Label htmlFor={id} {...(disabled ? { 'data-disabled': true } : {})}>
+        {label}
+      </Label>
+    </OptionWrapper>
+  );
+}
+
+const RadioGroupWrapper = styled(RadioGroup.Root)``;
+
+const OptionWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  margin: var(--size-spacing-sm) 0;
+`;
+
+const Input = styled(RadioGroup.Item)`
+  cursor: pointer;
+
+  box-sizing: border-box;
+  border: 1px solid var(--radio-fg);
+  border-radius: 100%;
+
+  height: var(--size-dimension-md);
+  width: var(--size-dimension-md);
+  min-width: var(--size-dimension-md);
+
+  &:hover,
+  &[data-state='checked'] {
+    background-color: var(--radio-bg);
+  }
+
+  &:focus,
+  &:focus-visible {
+    background-color: var(--radio-bg);
+    outline: var(--radio-bg) solid 4px;
+  }
+
+  &:disabled {
+    cursor: default;
+    background-color: var(--radio-bg-disabled);
+    border-color: var(--radio-fg-disabled);
+  }
+`;
+
+const SelectionIndicator = styled(RadioGroup.Indicator)`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
+  position: relative;
+
+  &::after {
+    content: '';
+    display: block;
+
+    box-sizing: border-box;
+    height: var(--radio-size-indicator);
+    width: var(--radio-size-indicator);
+    border-radius: 50%;
```

Comment:
Do we have design token for radius-full ? we probably should

Reply:
Yeah, let's keep that as a fix for when we have all the right tokens for the component